### PR TITLE
Fix all email notifications

### PR DESCRIPTION
**Pull request by mertcelebi**
Ah, what a journey this PR has been... Let me explain everything.

My first step in trying to fix the email issue on Tower was reaching out to Seqera Labs and pinging the Nextflow Slacks. Sadly, they were not super helpful, mainly because Seqera said "this works on Tower on our end." and Nextflow Slacks didn't  respond. It happens.
* On Tower, the issue seems like `sendmail` or `mail` utils are not present on the base AWS Batch ECS image. I could probably create my own AMI, but that seems like an unnecessary source of later technical debt.

My initial suspicion was that our SMTP secrets were not getting through. In order to prevent that, I hardcoded those values to make sure they were getting through. That was not an issue, but I realized for certain SMTP providers, you also have to input a `FROM` address, because otherwise Nextflow defaults to the instance address. Along the way, I also noticed a typo with SMTP vs SMPT. All fixed now. 

Seems like the existing `sendmail` and `mail` implementation that comes with the nf-core templates is quite old. And there are [many](https://nfcore.slack.com/archives/CE6SDBX2A/p1651159895275029) [many](https://nfcore.slack.com/archives/CE6SDBX2A/p1560364980084400) [many](https://nfcore.slack.com/archives/CE6SDBX2A/p1667418699029889) people struggle to get this working. 

Turns out, Nextflow actually has its own `sendMail` utility and it's been an [age-long issue](https://nextflow.slack.com/archives/C03R4B499CY/p1588510241090700) to switch to using that in nf-core.

Sadly, it was almost impossible to find any examples of people making this work using Nextflow's `sendMail`. This is most likely because the existing nf-core version worked decently well on HPCs (many comments about this). But the new built-in method is supposed to work across the spectrum. I thought this would be a win! So I decided to go at it alone. 

This led me to spelunk into Nextflow's repository to understand how their built-in `sendMail` works. That was difficult to fully figure out, because
* I didn't know in which contexts I could call built-in methods (turns out it works in `onComplete` but doesn't in any of the classes under `lib`.
* I didn't know what the right arguments were due to poor documentation. There are many places with conflicting instructions as to specifying the body of the email (`body` vs `content`, first is correct) and the attachments (`attach` vs `attachment` vs `attachments`, first one is correct).

Once I figured out everything, I tested it all end-to-end on an EC2 instance and it all seems to work. I need to do further testing on Tower, so please don't merge yet. As a nice side benefit of this, I got to delete bunch of template code!

Finally, I've attempted some stuff that didn't work out on Tower (like using Nextflow's `tempFile`s. The commit history of this PR explains it all.

### CODE REVIEW

File: conf/services.config [hunk 1]
```diff
@@ -16,16 +16,17 @@
     system (eg. sendmail or mail)."
 
     In the environments spun by Nextflow Tower on AWS Batch, the underlying
-    sendmail or mail calls are not available, so you have to specify SMPT
+    sendmail or mail calls are not available, so you have to specify SMTP
```
Comment (taylorreiter) [inline conf/services.config:19]: were these just typos, or is SMPT different from SMTP?

File: lib/NfcoreTemplate.groovy [hunk 2]
```diff
@@ -116,52 +121,15 @@ class NfcoreTemplate {
...
         }
 
-        // Render the TXT template
-        def engine       = new groovy.text.GStringTemplateEngine()
-        def tf           = new File("$projectDir/assets/email_template.txt")
-        def txt_template = engine.createTemplate(tf).make(email_fields)
-        def email_txt    = txt_template.toString()
-
         // Render the HTML template
+        def engine       = new groovy.text.GStringTemplateEngine()
```
Comment (taylorreiter) [inline lib/NfcoreTemplate.groovy]:
```suggestion
        def engine        = new groovy.text.GStringTemplateEngine()
```

File: workflows/seqqc.nf [hunk 3]
```diff
@@ -176,7 +176,32 @@ workflow SEQQC {
...
-        NfcoreTemplate.email(workflow, params, summary_params, projectDir, log, multiqc_report)
+        def email_params = NfcoreTemplate.get_email_params(workflow, params, summary_params, projectDir, log, multiqc_report)
+
+        // Note: this is a mild hack, because sendMail does not work with S3 URIs.
+        // First, we attempt to send our email regularly. If email_params.mqcFile is
+        // a local file path (ie you're running this pipeline on an HPC or locally),
+        // this will succeed. If it refers to an S3 object, this would fail.
+        // When it fails, we can forgo the main body of the email (mostly pipeline metadata)
+        // and try make MultiQC HTML (which already has some of the metadata), the
+        // body of our email.
```
Comment (taylorreiter) [inline workflows/seqqc.nf:187]: oh nice

File: workflows/seqqc.nf [hunk 4]
```diff
@@ -176,7 +176,32 @@ workflow SEQQC {
...
+                to: email_params.to,
+                subject: email_params.subject,
+                body: email_params.email_html,
+                attach: email_params.mqcFile
+            )
+        } catch (Exception e) {
+            sendMail (
+                to: email_params.to,
+                subject: email_params.subject,
+                body: email_params.mqcFile.text
```
Comment (taylorreiter) [inline workflows/seqqc.nf:206]: so the html text itself is the body of the email instead of an attachment?

File: workflows/seqqc.nf [hunk 5]
```diff
@@ -176,7 +176,32 @@ workflow SEQQC {
 
 workflow.onComplete {
     if (params.email || params.email_on_fail) {
-        NfcoreTemplate.email(workflow, params, summary_params, projectDir, log, multiqc_report)
+        def email_params = NfcoreTemplate.get_email_params(workflow, params, summary_params, projectDir, log, multiqc_report)
```
Comment (taylorreiter) [inline workflows/seqqc.nf:179]: ok so still using nf-core email templating, but not using nf-core send mail stuff

**Review by taylorreiter — APPROVED** [review]
I think I understand what's happening here (to an extent), but I don't think my review would catch problems as I don't understand all this stuff super well. You initial PR comment was super helpful in laying out an overview of everything though, so thank you!

I'm going to approve for now, lmk if you want a re-review if there are more changes

File: workflows/seqqc.nf (hunk 5 above)
Comment (mertcelebi) [reply workflows/seqqc.nf:179]: yes

File: workflows/seqqc.nf (hunk 4 above)
Comment (mertcelebi) [reply workflows/seqqc.nf:206]: yes

File: conf/services.config (hunk 1 above)
Comment (mertcelebi) [reply conf/services.config:19]: typos

File: workflows/seqqc.nf (hunk 4 above)
Comment (taylorreiter) [reply workflows/seqqc.nf:206]: cool, didn't know that was a thing!

**Review by taylorreiter — APPROVED** [review]
LGTM :) so exciting!

**Pull request merged**